libs/sdk-ui/src/base/measureFormats/fillMissingFormats.ts

```diff
@@ -0,0 +1,24 @@
+// (C) 2020-2022 GoodData Corporation
+import { IBucket, IInsightDefinition, insightBuckets, insightSetBuckets } from "@gooddata/sdk-model";
+import { fillMissingFormat } from "./fillMissingFormat";
+
+/**
+ * The function fills the format of the measures that does not have it set.
+ *
+ * @param insight - insight or insight definition that must be processed.
+ *
+ * @returns a copy of insight with auto-generated format for measures
+ *
+ * @internal
+ */
+export function fillMissingFormats<T extends IInsightDefinition>(insight: T): T {
+    const buckets = insightBuckets(insight);
+    return insightSetBuckets(insight, fillsMeasuresFormat(buckets));
+}
+
+function fillsMeasuresFormat(buckets: IBucket[]): IBucket[] {
+    return buckets.map((bucket) => ({
+        ...bucket,
```

Comment:
You could use `bucketModifyItems` from sdk-model, it would make this more maintainable.